doc/subsystems/conf_s11n/conf_s11n.rst

```diff
@@ -0,0 +1,156 @@
+.. _conf_s11n
+
+Runtime configuration system
+############################
+
+Config subsystem is meant for persisting per-device configuration
+and runtime state for modules.
+
+Configuration items are stored as key-value pairs, where both the key and
+the value are expected to be strings. Keys are divided into component
+elements, where packages register their subtree using the first element.
+E.g key ``id/serial`` consists of 2 components, ``id`` and ``serial``.
+Package sys/id registered it's subtree of configuration elements to be
+under ``id``.
+
+There are convenience routines for converting value back and forth
```

Comment:
How about:

    Convenience routines are provided for converting a key value to
    and from a string type.